tests/TestCase/ORM/TableRegistryTest.php

```diff
@@ -376,14 +368,23 @@ public function testRemove()
         $pluginTable = TableRegistry::get('TestPlugin.Comments');
         $cachedTable = TableRegistry::get('Comments');
 
+        $this->assertTrue(TableRegistry::exists('TestPlugin.Comments'));
+        $this->assertTrue(TableRegistry::exists('Comments'));
+        $this->assertNotSame($pluginTable, $cachedTable);
+
+        TableRegistry::remove('TestPlugin.Comments');
+        $this->assertFalse(TableRegistry::exists('TestPlugin.Comments'));
         $this->assertTrue(TableRegistry::exists('Comments'));
-        $this->assertSame($pluginTable, $cachedTable);
 
         TableRegistry::remove('Comments');
+        $this->assertFalse(TableRegistry::exists('TestPlugin.Comments'));
         $this->assertFalse(TableRegistry::exists('Comments'));
 
-        $appTable = TableRegistry::get('Comments');
-        $this->assertTrue(TableRegistry::exists('Comments'));
-        $this->assertNotSame($pluginTable, $appTable);
+        $pluginTable = TableRegistry::get('TestPlugin.Comments');
+        $cachedTable = TableRegistry::get('Comments');
```

Comment:
I think I'll move the test/asserts around to be more descriptive - rather than the current "make it work" state.